graphql-server/src/schemas/neo4j.ts

```diff
@@ -283,94 +331,177 @@ function buildResolvers(
       },
       consentPartC: (parent: PatientsListQuery["patients"][number]) => {
-        try {
-          return JSON.parse(
-            parent.hasSampleSamples[0].hasMetadataSampleMetadata[0]
-              .additionalProperties
-          )["consent-partc"];
-        } catch {
-          return undefined;
-        }
+        return getNestedValue(parent, "Patient", "consentPartC");
       },
     },
     Cohort: {
       totalSampleCount: (parent: CohortsListQuery["cohorts"][number]) => {
-        return parent.hasCohortSampleSamplesConnection?.totalCount;
+        return getNestedValue(parent, "Cohort", "totalSampleCount");
       },
       smileSampleIds: (parent: CohortsListQuery["cohorts"][number]) => {
-        return parent.hasCohortSampleSamples?.map((s) => s.smileSampleId);
+        return getNestedValue(parent, "Cohort", "smileSampleIds");
       },
       billed: (parent: CohortsListQuery["cohorts"][number]) => {
-        const samples = parent.hasCohortSampleSamples;
-        const allSamplesBilled =
-          samples?.length > 0 &&
-          samples.every((sample) => sample.hasTempoTempos?.[0]?.billed);
-        return allSamplesBilled ? "Yes" : "No";
+        return getNestedValue(parent, "Cohort", "billed");
       },
       initialCohortDeliveryDate: (
         parent: CohortsListQuery["cohorts"][number]
       ) => {
-        return parent.hasCohortCompleteCohortCompletes?.slice(-1)[0]?.date;
+        return getNestedValue(parent, "Cohort", "initialCohortDeliveryDate");
       },
       endUsers: (parent: CohortsListQuery["cohorts"][number]) => {
-        return parent.hasCohortCompleteCohortCompletes?.[0]?.endUsers; // latest
+        return getNestedValue(parent, "Cohort", "endUsers");
       },
       pmUsers: (parent: CohortsListQuery["cohorts"][number]) => {
-        return parent.hasCohortCompleteCohortCompletes?.[0]?.pmUsers;
+        return getNestedValue(parent, "Cohort", "pmUsers");
       },
       projectTitle: (parent: CohortsListQuery["cohorts"][number]) => {
-        return parent.hasCohortCompleteCohortCompletes?.[0]?.projectTitle;
+        return getNestedValue(parent, "Cohort", "projectTitle");
       },
       projectSubtitle: (parent: CohortsListQuery["cohorts"][number]) => {
-        return parent.hasCohortCompleteCohortCompletes?.[0]?.projectSubtitle;
+        return getNestedValue(parent, "Cohort", "projectSubtitle");
       },
       status: (parent: CohortsListQuery["cohorts"][number]) => {
-        return parent.hasCohortCompleteCohortCompletes?.[0]?.status;
+        return getNestedValue(parent, "Cohort", "status");
       },
       type: (parent: CohortsListQuery["cohorts"][number]) => {
-        return parent.hasCohortCompleteCohortCompletes?.[0]?.type;
+        return getNestedValue(parent, "Cohort", "type");
       },
     },
   };
 }
 
+function sortArrayByNestedField(
+  arr: any[],
+  nodeLabel: string,
+  fieldName: string,
+  sortOrder: SortDirection
+) {
+  arr.sort((objA, objB) => {
```

Comment:
Re my comment [here](https://github.com/mskcc/smile-dashboard/pull/143/files#diff-db966dd79103f32fddb49dcefcfba78d4325506ce364cd0e24779a3521aa7173R513) - maybe the logic _there_ can be moved to a function call also like how this function is set up?